app/javascript/App/bundles/CoinShow/InfoBar.tsx

```diff
@@ -138,112 +140,126 @@ class InfoBar extends React.Component<Props, {}> {
         : { color: negativeColour }
 
     return (
-      <Grid
-        container={true}
-        justify="flex-start"
-        alignItems="flex-start"
-        className={classes.root}
-      >
-        <Grid
-          item={true}
-          xs={12}
-          md={3}
-          container={true}
-          justify="flex-start"
-          alignItems="baseline"
-          alignContent="flex-start"
-        >
-          <Grid item={true} className={classes.coinImage}>
-            <img alt={coinObj.name} src={coinObj.image_url} />
-          </Grid>
-          <Grid item={true} className={classes.coinName}>
-            {coinObj.name}
-          </Grid>
-          <Grid item={true} className={classes.coinSymbol}>
-            {!_.isUndefined(symbol) && `(${symbol})`}
-          </Grid>
-          <Grid item={true} xs={12} style={{ height: '10px' }} />
-          <Grid item={true} className={classes.coinRanking}>
-            Rank #{ranking}
-          </Grid>
-          <Grid item={true} className={classes.watchButtonContainer}>
-            <Icon
-              name="star"
-              solid={true}
-              className={classnames(
-                classes.watchButton,
-                isWatched ? classes.watchedButton : classes.unwatchedButton,
-              )}
-              onClick={watchCoinHandler}
+      <CurrencyContext.Consumer>
```

Comment:
You should be able to use your new `withCurrency` HOC in all the areas that need `CurrencyContext.Consumer`

Reply:
Yeah I'm aware, but didn't really feel like converting them all since it wasn't really necessary.

Reply:
I guess we can just slowly migrate those components across